services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/notifications/ConnectionOfTravelNotification/ConnectionOfTravelNotificationRepositoryImpl.java

```diff
@@ -1,79 +1,86 @@
 
 @Component
-public class ConnectionOfTravelNotificationRepositoryImpl implements ConnectionOfTravelNotificationRepository {
+public class ConnectionOfTravelNotificationRepositoryImpl
+        implements ConnectionOfTravelNotificationRepository, PageWrapper, PaginatedQueryInterface {
 
     private final MongoTemplate mongoTemplate;
 
-    @Value("${maximumResponseSize}")
-    int maximumResponseSize;
-
     private final String collectionName = "CmConnectionOfTravelNotification";
+    private final String DATE_FIELD = "notificationGeneratedAt";
+    private final String INTERSECTION_ID_FIELD = "IntersectionID";
 
     @Autowired
     public ConnectionOfTravelNotificationRepositoryImpl(MongoTemplate mongoTemplate) {
         this.mongoTemplate = mongoTemplate;
     }
 
-    public Query getQuery(Integer intersectionID, Long startTime, Long endTime, boolean latest) {
-        Query query = new Query();
-
-        if (intersectionID != null) {
-            query.addCriteria(Criteria.where("intersectionID").is(intersectionID));
-        }
-
-        Date startTimeDate = new Date(0);
-        Date endTimeDate = new Date();
-
-        if (startTime != null) {
-            startTimeDate = new Date(startTime);
-        }
-        if (endTime != null) {
-            endTimeDate = new Date(endTime);
-        }
-
-        query.addCriteria(Criteria.where("notificationGeneratedAt").gte(startTimeDate).lte(endTimeDate));
-
-        if (latest) {
-            query.with(Sort.by(Sort.Direction.DESC, "notificationGeneratedAt"));
-            query.limit(1);
-        } else {
-            query.limit(maximumResponseSize);
-        }
-        return query;
+    public long getQueryResultCount(
+            Integer intersectionID,
+            Long startTime,
+            Long endTime,
+            Pageable pageable) {
+        Criteria criteria = new IntersectionCriteria()
+                .whereOptional(INTERSECTION_ID_FIELD, intersectionID)
+                .withinTimeWindow(DATE_FIELD, startTime, endTime)
+                .build();
+        return mongoTemplate.count(Query
+                .query(criteria).with(pageable), collectionName);
     }
 
-    public long getQueryResultCount(Query query) {
-        return mongoTemplate.count(query, ConnectionOfTravelNotification.class, collectionName);
+    public Page<ConnectionOfTravelNotification> findLatest(
+            Integer intersectionID,
+            Long startTime,
+            Long endTime) {
+        Criteria criteria = new IntersectionCriteria()
+                .whereOptional(INTERSECTION_ID_FIELD, intersectionID)
+                .withinTimeWindow(DATE_FIELD, startTime, endTime)
+                .build();
+        Sort sort = Sort.by(Sort.Direction.DESC, DATE_FIELD);
+        return wrapSingleResultWithPage(mongoTemplate.findOne(
+                Query.query(criteria).with(sort),
+                ConnectionOfTravelNotification.class,
+                collectionName));
```

Comment:
polish(non-blocking): with the amount of arguments involved in this method call, we could improve the reability of this section if we added some extra indentation. This makes it easier to understand that the results from `findOne` are passed into `wrapSingleResultWithPage`. It also makes it clearer that `wrapSingleResultWithPage` doesn't really care about the args passed into `findOne`. 

```suggestion
        return wrapSingleResultWithPage(
                mongoTemplate.findOne(
                        Query.query(criteria).with(sort),
                        ConnectionOfTravelNotification.class,
                        collectionName
                )
        );
```

Reply:
updated